File: frontend/src/types/analysis.ts
export type Verdict = "Safe" | "Suspicious" | "Malicious";
export type AnalysisStatus = "requested" | "running" | "completed" | "failed";

export interface AnalysisResponse {
  id: string;
  submitted_url: string;
  normalized_url: string | null;
  status: AnalysisStatus;
  verdict: Verdict | null;
  risk_score: number | null;
  explanation: string | null;
  evidence: Record<string, unknown>;
  error_message: string | null;
  created_at: string;
  updated_at: string;
  completed_at: string | null;
}

File: backend/src/clicksafe/domain/analysis.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Any

from clicksafe.domain.enums import AnalysisStatus, Verdict


@dataclass(slots=True)
class UrlAnalysisContext:
    submitted_url: str
    normalized_url: str | None = None
    final_url: str | None = None
    redirects: list[str] = field(default_factory=list)
    html: str | None = None
    screenshot_path: str | None = None
    metadata: dict[str, Any] = field(default_factory=dict)


@dataclass(frozen=True, slots=True)
class AnalysisJob:
    id: str
    submitted_url: str
    normalized_url: str | None
    status: AnalysisStatus
    verdict: Verdict | None
    risk_score: int | None
    explanation: str | None
    evidence: dict[str, Any]
    error_message: str | None
    created_at: datetime
    updated_at: datetime
    completed_at: datetime | None
